### PR TITLE
Fix matrix ide configs

### PR DESCRIPTION
**Pull request by Arrash03**

### CODE REVIEW

File: compile_commands.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+[
+{
+  "directory": "/home/fanaa/projects/other/Matrix/Release",
+  "command": "/usr/bin/g++ -O3 -DNDEBUG -std=c++20 -o src/CMakeFiles/Matrix.dir/main.cpp.o -c /home/fanaa/projects/other/Matrix/src/main.cpp",
```
Comment (SAMoosavi) [inline compile_commands.json]: it's your local address edit it

File: src/matrix-tmp.h [hunk 2]
```diff
@@ -68,8 +78,8 @@ Matrix<Element> Matrix<Element>::operator-() const
 {
 	Matrix tmp(*this);
 
-	for (RowType& row_of_tmp: tmp.table)
-		for (Element& element_of_tmp: row_of_tmp)
+	for (RowType& row_of_tmp : tmp.table)
```
Comment (SAMoosavi) [inline src/matrix-tmp.h:81]: This style is not good, like it: `for (RowType& row_of_tmp: tmp.table)`

File: src/polynomial-helper-tmp.h [hunk 3]
```diff
@@ -33,8 +32,6 @@ Int create_random_number(Int begin, Int end) noexcept
 	return distribution(gen);
 }
 
-
-}// namespace polynomial_helper
-
+}		 // namespace polynomial_helper
```
Comment (SAMoosavi) [inline src/polynomial-helper-tmp.h:35]: Don't use space before command

File: src/polynomial-tmp.h [hunk 4]
```diff
@@ -34,20 +34,24 @@ Polynomial<Element>::NewtonOutput Polynomial<Element>::solve_by_newton(double gu
...
+		if (compare_with_precision(derived_answer, 0.0, precision / 2))
+		{
 			// TODO: it could replace with 2 or three as precision. check this
 			if (compare_with_precision(polynomial_answer, 0.0, precision / 2))
 				result.second = true;
 			else
 				current_answer = polynomial_helper::create_random_number(static_cast<int64_t>(guess - guess * 0.1),
-				                                                         static_cast<int64_t>(guess + guess * 0.1));
-		} else {
+						static_cast<int64_t>(guess + guess * 0.1));
```
Comment (SAMoosavi) [inline src/polynomial-tmp.h:51]: Just use two tab

File: test/matrixFunctionality.cpp [hunk 5]
```diff
@@ -233,49 +216,23 @@ TEST_P(MultipleOfTwoMatrix, TheMultipleEqualOperatorShouldSetFirstMatrixEqualToT
...
-		)
-);
-
-class OppositeOfMatrix: public ::testing::TestWithParam<std::tuple<Matrix<int>, Matrix<int>>> {};
+INSTANTIATE_TEST_SUITE_P(MultipleData, MultipleOfTwoMatrix,
+		Values(std::make_tuple(Matrix<int>({{1, 2}, {1, 2}}), Matrix<int>({{1, 2}, {1, 2}}),
+					   Matrix<int>({{3, 6}, {3, 6}})),
+				std::make_tuple(Matrix<int>({{1, 2, 3, 4, 5, 6, 7}}), Matrix<int>({{7}, {6}, {5}, {4}, {3}, {2}, {1}}),
+						Matrix<int>({{84, 0, 0, 0, 0, 0, 0}})),
+				std::make_tuple(Matrix<int>({{1, 2, 3, 4, 5, 6, 7}, {1, 2, 3, 4, 5, 6, 7}, {1, 2, 3, 4, 5, 6, 7},
```
Comment (SAMoosavi) [inline test/matrixFunctionality.cpp:224]: Can we say to clang-format each nested array in a new line if need break?

File: compile_commands.json (hunk 1 above)
Comment (Arrash03) [reply compile_commands.json]: fixed

File: src/polynomial-helper-tmp.h (hunk 3 above)
Comment (Arrash03) [reply src/polynomial-helper-tmp.h:35]: clang-format added this

File: src/polynomial-tmp.h (hunk 4 above)
Comment (Arrash03) [reply src/polynomial-tmp.h:51]: why?

File: test/matrixFunctionality.cpp (hunk 5 above)
Comment (Arrash03) [reply test/matrixFunctionality.cpp:224]: i do not know

File: src/polynomial-helper-tmp.h (hunk 3 above)
Comment (SAMoosavi) [reply src/polynomial-helper-tmp.h:35]: edit clang format that don't add space

File: test/matrixFunctionality.cpp (hunk 5 above)
Comment (SAMoosavi) [reply test/matrixFunctionality.cpp:224]: search it :)

**Pull request merged**